### PR TITLE
Components

### PR DESCRIPTION
**Pull request by MuriloRoque**
Dear TSE, hello! :wave:  

Thank you for reviewing my project :grin:

Here's what I did in this milestone:

- Organized the directory structure;
- Created a components directory in the src folder;
- Created four files in there, each one representing one React component;
- Implemented the App component, rendering Display and ButtonPanel;
- Implemented the Display component, showing the result of the calculation as an incoming prop (a String with a default value of 0);
- Implemented the Button component, accepting a button name as a prop (a String);
- Implemented the ButtonPanel component, rendering the calculator panel with all the buttons in the following groups:
   - Group 1: AC, +/-, %, ÷
   - Group 2: 7, 8, 9, X
   - Group 3: 4, 5, 6, -
   - Group 4: 1, 2, 3, +
   - Group 5: 0, ., =
- Wrote some tests using Jest and Enzyme;
- Added some minimal styles;
- Fixed linter errors;
- Updated the deployment to Heroku;
- Updated the README file.

### CODE REVIEW

File: src/components/Display.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import '../styles/Display.css';
+import PropTypes from 'prop-types';
+
+const Display = props => {
```
Comment (shshamim63) [inline src/components/Display.js]: [optional]
You can destructure it in the props in the following way.
```js
const Display = { result } => {
```
Then you can use the result directly.

File: src/components/ButtonPanel.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      <Button name="2" />
+      <Button name="3" />
+      <Button name="+" />
+    </div>
+    <div className="row">
+      <Button name="0" />
+      <Button name="." />
+      <Button className="equal" name="=" />
+    </div>
+  </div>
```
Comment (shshamim63) [inline src/components/ButtonPanel.js]: [Optional]
There is a way to refactor this code. Think about an array containing multiple arrays filled with button names. Then you can use two loops to cover this scenario.

File: src/components/Button.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import '../styles/Button.css';
+import PropTypes from 'prop-types';
+
+const Button = props => {
```
Comment (shshamim63) [inline src/components/Button.js]: [Optional]
You can destructure the props element in here.

File: src/components/App.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import Display from './Display';
+import ButtonPanel from './ButtonPanel';
+
+const App = () => (
+  <div id="parent-div">
+    <Display result="0" />
```
Comment (shshamim63) [inline src/components/App.js]: The default value should not be included here. Remove the result assigning.

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## :recycle: Required Changes :recycle: 
___
Hi @MuriloRoque :wave: 
You almost completed the milestone :clap: , just found an (important and easy to fix) issue. Let's fix it.
___
### :spades: Mandatory Changes :spades: 
___
- You should add the default value of the result using defaultProps in the display component.
___
### :clubs: Optional Changes :clubs: 
___
- You can refactor the code a little bit more. Checkout the comments.
___
##### please fix the above-mentioned issue and send review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Comment by MuriloRoque** [comment]
@shshamim63 thank you for your nice suggestions :grin:

The changes have been made, I will wait for the next review.

**Review by Torres-ssf — APPROVED** [review]
### Milestone Approved :+1: 
Hi there @MuriloRoque 
Great job on this milestone :clap: 
Good luck with the next ones.

Happy Coding :computer: 
**Sergio Torres**
[Twitter](https://twitter.com/torres_ssf) [LinkedIn](https://www.linkedin.com/in/torres-ssf/) [Portifolio](https://www.torres-ssf.com/)

**Comment by MuriloRoque** [comment]
Thank you, @Torres-ssf :grin:

**Pull request merged**